netty/src/main/java/io/grpc/transport/netty/NettyClientHandler.java

```diff
@@ -136,21 +137,21 @@ public void channelActive(ChannelHandlerContext ctx) throws Exception {
    */
   @Override
   public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise) {
-    try {
-      if (msg instanceof CreateStreamCommand) {
-        createStream((CreateStreamCommand) msg, promise);
-      } else if (msg instanceof SendGrpcFrameCommand) {
-        sendGrpcFrame(ctx, (SendGrpcFrameCommand) msg, promise);
-      } else if (msg instanceof CancelStreamCommand) {
-        cancelStream(ctx, (CancelStreamCommand) msg, promise);
-      } else {
-        throw new AssertionError("Write called for unexpected type: " + msg.getClass().getName());
-      }
-    } catch (Throwable t) {
-      promise.setFailure(t);
+    if (msg instanceof ClientWriteQueue) {
+      ((ClientWriteQueue) msg).flush(ctx, promise);
+    } else {
+      throw new AssertionError("Write called for unexpected type: " + msg.getClass().getName());
     }
   }
 
+  void startWriteQueue(Channel channel) {
```

Comment:
Do we need this?  If we created the write queue in `handlerAdded` that should be early enough. WDYT?